src/renderer/components/archive/archive-search.js

```diff
@@ -177,8 +177,11 @@ class ArchiveSearch extends PureComponent {
    * @memberof ArchiveSearch
    */
   highlightSearchResult = word => {
-    const regex = new RegExp('(' + this.state.searchTerm + ')', 'ig');
-    return word.replace(regex, '<mark>$1</mark>');
+    try {
+      const regex = new RegExp('(' + this.state.searchTerm + ')', 'ig');
+      return word.replace(regex, '<mark>$1</mark>');
+    } catch(e) {}
+    return word;
```

Comment:
Good idea! Though I'd recommend making this simpler now that we're looking at it. Using a regex is obviously going to cause issues when special chars are used, so we should opt for plain text.

Simply finding the index and replacing the word from that point only should do the trick What about something like this:

```javascript
const index = word.toLowerCase().indexOf(this.state.searchTerm.toLowerCase());
if (index >= 0) {
  return `${word.substring(0, index)}<mark>${this.state.searchTerm}</mark>${word.substring(index + this.state.searchTerm.length)}`;
}
return word;
```

Reply:
I haven't tested if this works, but it'd do away with the complexity of regular expressions.

Reply:
How about escaping any `RegExp` characters in the search term using [escape-string-regexp](https://www.npmjs.com/package/escape-string-regexp)?

Reply:
We can avoid the dependency by copy-pasting.

```js

  highlightSearchResult = word => {
    const searchTerm = this.state.searchTerm.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&');
    const regex = new RegExp('(' + searchTerm + ')', 'ig');
    return word.replace(regex, '<mark>$1</mark>');
  };
```

Reply:
Pushed. 😄

Reply:
Sorry I didn't reply sooner. It's still not as robust as what I suggested, I'm afraid. There's no need for regex at all in this situation. It's a simple result of highlighting the first instance of the search term, which my example does with only string splitting.